addons/stock/models/stock_orderpoint.py

```diff
@@ -179,6 +179,17 @@ def write(self, vals):
                     raise UserError(_("Changing the company of this record is forbidden at this point, you should rather archive it and create a new one."))
         return super().write(vals)
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        if isinstance(vals_list, dict):
+            vals_list = [vals_list]
+        for vals in vals_list:
+            if vals.get('product_id', False):
+                product = self.env['product.product'].browse(vals['product_id'])
+                if product.type != 'product':
+                    raise ValidationError(_('You cannot create a replenishment rule for a non-storable product.'))
+        return super().create(vals_list)
```

Comment:
This whole thing should be done through a `@api.constrains()` method. Something similar is done in `stock_quant.py`.

Reply:
Makes sense 👍